BrickKit/app/src/main/java/com/wayfair/brickkitdemo/bricks/UsedBrick.java

```diff
@@ -59,10 +59,12 @@ public void setText(CharSequence text) {
 
     @Override
     public void onBindData(final BrickViewHolder viewHolder) {
-        UsedBrickHolder holder = (UsedBrickHolder) viewHolder;
-        holder.textView.setText(text);
-        if (isEnabled()) {
-            holder.itemView.setOnClickListener(onTouch());
+        if(viewHolder instanceof UsedBrickHolder) {
+            UsedBrickHolder holder = (UsedBrickHolder) viewHolder;
+            holder.textView.setText(text);
+            if (isEnabled()) {
```

Comment:
do you have to clear the previous clicklistener if the view is being recycled?

Reply:
I fixed.  Good find. 👍